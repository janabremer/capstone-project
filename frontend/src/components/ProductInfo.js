import styled from 'styled-components/macro'
import PropTypes from 'prop-types'

ProductInfo.propTypes = {
    name: PropTypes.string,
    water: PropTypes.number,
    requestStatus: PropTypes.string
}

export default function ProductInfo ({name, water, requestStatus}) {
    switch (requestStatus) {
        case 'LOADING':
            return(
                <ProductInfoStyled>
                    <Content>loading...</Content>
                </ProductInfoStyled>
            )
        case 'ERROR':
            return(
                <ProductInfoStyled>
                    <Content>error: server is down</Content>
                </ProductInfoStyled>
            )
        default:
            return(
                <ProductInfoStyled>
                    <Heading>{name}</Heading>
                    <Content>{water} litres / kg</Content>
                </ProductInfoStyled>
            )
    }
}

const ProductInfoStyled = styled.section`
    background-color: var(--primary-light-transparent);
    display: grid;
    gap: var(--gap-small);
    padding: 15px 0;
    place-items: center;
    width: 100%;
`

const Heading = styled.h1`
    text-transform: uppercase;
    font-size: 170%;
    letter-spacing: 0.25rem;
    font-weight: 700;
`

const Content = styled.span`
    font-size: 120%;
    color: var(--secondary-dark);
`