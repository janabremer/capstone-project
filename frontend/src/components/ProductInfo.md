```js
<ProductInfo name={'apple'} water={'822'} />
```
