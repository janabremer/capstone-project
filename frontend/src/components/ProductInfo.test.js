import { render } from '@testing-library/react'
import 'jest-styled-components'
import ProductInfo from './ProductInfo'

const testProps = {
    name: 'apple',
    water: 822,
}

describe('ProductInfo', () => {
    it('renders correctly', () => {
        const { container } = render(
            <ProductInfo {...testProps} />
        )
        expect(container.firstChild).toMatchSnapshot()
    })

    it('displays product name and product water', () => {
        const { getByText } = render(
            <ProductInfo {...testProps} />
        )
        expect(getByText(/apple/i)).toBeInTheDocument()
        expect(getByText(/822/i)).toBeInTheDocument()
    })
    
})