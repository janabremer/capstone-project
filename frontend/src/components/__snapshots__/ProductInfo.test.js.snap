// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductInfo renders correctly 1`] = `
.c0 {
  background-color: var(--primary-light-transparent);
  display: grid;
  gap: var(--gap-small);
  padding: 15px 0;
  place-items: center;
  width: 100%;
}

.c1 {
  text-transform: uppercase;
  font-size: 170%;
  -webkit-letter-spacing: 0.25rem;
  -moz-letter-spacing: 0.25rem;
  -ms-letter-spacing: 0.25rem;
  letter-spacing: 0.25rem;
  font-weight: 700;
}

.c2 {
  font-size: 120%;
  color: var(--secondary-dark);
}

<section
  class="c0"
>
  <h1
    class="c1"
  >
    apple
  </h1>
  <span
    class="c2"
  >
    822
     litres / kg
  </span>
</section>
`;
